twilight-model/src/application/interaction/mod.rs

```diff
@@ -464,6 +477,22 @@ pub enum InteractionData {
     ModalSubmit(ModalInteractionData),
 }
 
+/// Partial guild containing only the fields sent in the partial guild
+/// in interactions.
+///
+/// # Note that the field `locale` does not exists on the full guild
+/// object, and is only found here. See
+/// <https://github.com/discord/discord-api-docs/issues/6938> for more
+/// info.
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize, Hash)]
+pub struct InteractionPartialGuild {
+    /// Id of the guild.
+    pub id: Option<Id<GuildMarker>>,
+    /// Enabled guild features
+    pub features: Option<Vec<GuildFeature>>,
+    pub locale: Option<String>,
+}
+
```

Comment:
Probably those fields are guaranteed to exist for interactions, but it's not documented and I'm not 100% sure about that. ig option is fine.

Reply:
Yeah this partial guild is strange so I will rather go the safe route. For example the guild id is sent in the interaction so maybe they will stop duplication it here.